### PR TITLE
Update UFSDA ATM anl for new COM directory structure

### LINKED ISSUE #1517: Update UFSDA ATM anl for COM refactor
**Description**
UFSDA jobs were not included in the initial COM refactor work (#1421). These jobs will all need to be updated to use the new structure and templates before they can run again.

**Requirements**
UFSDA deterministic ATM analysis jobs that utilize the new COM templates to generate COM directory paths.

**Acceptance Criteria (Definition of Done)**
UFSDA deterministic ATM analysis jobs that work as well as before the COM refactor

**(Optional): Suggest A Solution**
There are two approaches we can take. The first is to define the full paths in the j-job using the templates, as is done for most of the existing global-workflow jobs, and update the python to expect full paths (with dates and cycles). The second is to use the templates directly in the python (reading the `*_TMPL` variables from the environment) and using Rahul's Template class to fill in the variables.

Recommend the first for now where feasible, though the latter may be needed if iterating over members in python (similar to how the EnKF exscripts have to do it). The Template class will need a minor update to proceed with option 2: The templates reply on the bash substitution replacing undefined variables with an empty string (mainly to replacing `$MEMDIR` for the deterministic files), but the class currently leaves any variables it doesn't have substitutions for as-is. It will need to be updated to provide the option to replace undefined variables with an empty string.

### PR DESCRIPTION
**Pull request by RussTreadon-NOAA**
**Description**
g-w PR #1421 changed the GFS COM directory structure.   This PR updates UFSDA ATM variational analysis jobs and python scripts to work with the updated GFS COM directory structure.

Fixes #1517

**Type of change**
- [x] Bug fix (non-breaking change which fixes an issue)

**How Has This Been Tested?**
- [x] Clone and Build tests on Hera
- [x] Cycled test on Hera
  
**Checklist**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Comment by RussTreadon-NOAA** [comment]
@WalterKolczynski-NOAA , the linter failure is
```
✋ Defects, NEEDS INSPECTION
Error: SHELLCHECK_WARNING:
./jobs/JGLOBAL_ATM_ANALYSIS_FINALIZE:11:28: note[SC2153]: Possible misspelling: PDY may not be assigned. Did you mean gPDY?

Error: SHELLCHECK_WARNING:
./jobs/JGLOBAL_ATM_ANALYSIS_INITIALIZE:10:28: note[SC2153]: Possible misspelling: PDY may not be assigned. Did you mean gPDY?
```
For both jobs the offending line is

` 
GDATE=$(date +%Y%m%d%H -d "${PDY} ${cyc} - ${assim_freq} hours")
 `

Both the initialize and finalize jobs use files from the previous cycle.    The rocoto workflow defines `PDY`.

What is the preferred solution to resolve the linter failure?

File: jobs/JGLOBAL_ATM_ANALYSIS_INITIALIZE [hunk 1]
```diff
@@ -8,19 +8,29 @@ source "${HOMEgfs}/ush/jjob_header.sh" -e "atmanlinit" -c "base atmanl atmanlini
 # Set variables used in the script
 ##############################################
 GDATE=$(date +%Y%m%d%H -d "${PDY} ${cyc} - ${assim_freq} hours")
+export gPDY=${GDATE:0:8}
+export gcyc=${GDATE:8:2}
```
Comment (aerorahul) [inline jobs/JGLOBAL_ATM_ANALYSIS_INITIALIZE]: do we need to export these?

File: jobs/JGLOBAL_ATM_ANALYSIS_INITIALIZE [hunk 2]
```diff
@@ -8,19 +8,29 @@ source "${HOMEgfs}/ush/jjob_header.sh" -e "atmanlinit" -c "base atmanl atmanlini
...
 GDUMP="gdas"
+export GDUMP_ENS="enkf${GDUMP}"
```
Comment (aerorahul) [inline jobs/JGLOBAL_ATM_ANALYSIS_INITIALIZE]: does this need to be exported?

File: jobs/JGLOBAL_ATM_ANALYSIS_FINALIZE [hunk 3]
```diff
@@ -9,19 +9,25 @@ source "${HOMEgfs}/ush/jjob_header.sh" -e "atmanlfinal" -c "base atmanl atmanlfi
 # Set variables used in the script
 ##############################################
 GDATE=$(date +%Y%m%d%H -d "${PDY} ${cyc} - ${assim_freq} hours")
+export gPDY=${GDATE:0:8}
+export gcyc=${GDATE:8:2}
```
Comment (aerorahul) [inline jobs/JGLOBAL_ATM_ANALYSIS_FINALIZE]: do these need to be exported?

**Review by aerorahul — CHANGES_REQUESTED** [review]
I am glad the updates after the reorg were fairly straightforward.

Looks good, with a couple of questions.

**Comment by aerorahul** [comment]
> @WalterKolczynski-NOAA , the linter failure is
> 
> ```
> ✋ Defects, NEEDS INSPECTION
> Error: SHELLCHECK_WARNING:
> ./jobs/JGLOBAL_ATM_ANALYSIS_FINALIZE:11:28: note[SC2153]: Possible misspelling: PDY may not be assigned. Did you mean gPDY?
> 
> Error: SHELLCHECK_WARNING:
> ./jobs/JGLOBAL_ATM_ANALYSIS_INITIALIZE:10:28: note[SC2153]: Possible misspelling: PDY may not be assigned. Did you mean gPDY?
> ```
> 
> For both jobs the offending line is
> 
> ` GDATE=$(date +%Y%m%d%H -d "${PDY} ${cyc} - ${assim_freq} hours")`
> 
> Both the initialize and finalize jobs use files from the previous cycle. The rocoto workflow defines `PDY`.
> 
> What is the preferred solution to resolve the linter failure?

place a line like this before:
```
# shellcheck disable=SC2153
```

File: jobs/JGLOBAL_ATM_ANALYSIS_INITIALIZE (hunk 1 above)
Comment (RussTreadon-NOAA) [reply jobs/JGLOBAL_ATM_ANALYSIS_INITIALIZE]: I simply followed what's done in `JGLOBAL_ATMOS_ANALYSIS`.  I'm fine with removing the export.

File: jobs/JGLOBAL_ATM_ANALYSIS_INITIALIZE (hunk 2 above)
Comment (RussTreadon-NOAA) [reply jobs/JGLOBAL_ATM_ANALYSIS_INITIALIZE]: same reply as above.

File: jobs/JGLOBAL_ATM_ANALYSIS_FINALIZE (hunk 3 above)
Comment (RussTreadon-NOAA) [reply jobs/JGLOBAL_ATM_ANALYSIS_FINALIZE]: same reply as above.

File: jobs/JGLOBAL_ATM_ANALYSIS_INITIALIZE (hunk 1 above)
Comment (aerorahul) [reply jobs/JGLOBAL_ATM_ANALYSIS_INITIALIZE]: we shouldn't export variables not needed by child scripts.

Comment (RussTreadon-NOAA) [reply jobs/JGLOBAL_ATM_ANALYSIS_INITIALIZE]: export removed

File: jobs/JGLOBAL_ATM_ANALYSIS_INITIALIZE (hunk 2 above)
Comment (RussTreadon-NOAA) [reply jobs/JGLOBAL_ATM_ANALYSIS_INITIALIZE]: export removed

File: jobs/JGLOBAL_ATM_ANALYSIS_FINALIZE (hunk 3 above)
Comment (RussTreadon-NOAA) [reply jobs/JGLOBAL_ATM_ANALYSIS_FINALIZE]: export removed

**Review by aerorahul — APPROVED** [review]
Looks great.
Thanks!

**Comment by RussTreadon-NOAA** [comment]
Thank you @aerorahul for the quick review, guidance, and approval.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**